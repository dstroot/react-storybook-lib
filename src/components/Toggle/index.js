import React from 'react';
import PropTypes from 'prop-types';

// styles
import './styles.scss';

/**
  Toggle is a component ...
 */

const Toggle = ({ isOn, handleToggle, onColor }) => {
  return (
    <>
      <input
        checked={isOn}
        onChange={handleToggle}
        className="react-switch-checkbox"
        id={`react-switch-new`}
        type="checkbox"
      />
      <label
        className="react-switch-label"
        htmlFor={`react-switch-new`}
        style={{ background: isOn && onColor }}
      >
        <span className={`react-switch-button`} />
      </label>
    </>
  );
};

Toggle.propTypes = {
  /** Boolean to termine of the toggle is on or off */
  isOn: PropTypes.bool.isRequired,
  /** Function to perform an action due to toggle */
  handleToggle: PropTypes.func.isRequired,
  /** Color for switched on view */
  onColor: PropTypes.string,
};

export default Toggle;
