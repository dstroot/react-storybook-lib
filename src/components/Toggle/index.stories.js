import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';

// components
import Toggle from '../Toggle';

function useToggle(initialValue) {
  const [value, setValue] = useState(initialValue);
  const toggleValue = () => setValue(!value);
  return [value, toggleValue];
}

// Button section
storiesOf('VDS|Toggle', module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false,
    },
  })
  .add('show toggle off', () => {
    const [isOn, toggleFn] = useToggle(false);
    return <Toggle isOn={isOn} handleToggle={toggleFn} onColor="#EF476F" />;
  })
  .add('show toggle on', () => {
    const [isOn, toggleFn] = useToggle(true);
    return <Toggle isOn={isOn} handleToggle={toggleFn} onColor="#06D6A0" />;
  });
