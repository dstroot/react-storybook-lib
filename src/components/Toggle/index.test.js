import React from 'react';
import Toggle from '../Toggle';
import renderer from 'react-test-renderer';

// don't do this - its just to make Toggle render
let value = true;
const toggle = () => {
  value = !value;
};

test('Toggle renders a toggle switch - OFF', () => {
  const component = renderer.create(
    <Toggle isOn={false} handleToggle={toggle} onColor="blue" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Toggle renders a toggle switch - ON', () => {
  const component = renderer.create(
    <Toggle isOn={true} handleToggle={toggle} onColor="blue" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
