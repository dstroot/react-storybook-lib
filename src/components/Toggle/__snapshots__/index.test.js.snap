// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders a toggle switch - OFF 1`] = `
Array [
  <input
    checked={false}
    className="react-switch-checkbox"
    id="react-switch-new"
    onChange={[Function]}
    type="checkbox"
  />,
  <label
    className="react-switch-label"
    htmlFor="react-switch-new"
    style={
      Object {
        "background": false,
      }
    }
  >
    <span
      className="react-switch-button"
    />
  </label>,
]
`;

exports[`Toggle renders a toggle switch - ON 1`] = `
Array [
  <input
    checked={true}
    className="react-switch-checkbox"
    id="react-switch-new"
    onChange={[Function]}
    type="checkbox"
  />,
  <label
    className="react-switch-label"
    htmlFor="react-switch-new"
    style={
      Object {
        "background": undefined,
      }
    }
  >
    <span
      className="react-switch-button"
    />
  </label>,
]
`;
